### PR TITLE
Change Readme Steps and add example config files

### PR DESCRIPTION
**Pull request by aganesh-suse**
Changes involved: 
1. Added steps in README.md file:
    a. Setup changes to get the test runs started in local environments.
    b. Help for macOS M2 chip machine to run tests.
2. Added following files to config/examples folder: 
   a. k3s.tfvars file
   b. rke2.tfvars file
   c. config.yaml file

https://github.com/rancher/qa-tasks/issues/587

### CODE REVIEW

File: config/k3s.tfvars.example
Comment (mdrahman-suse) [inline config/k3s.tfvars.example:22]: aws_ami info should not be listed here

File: config/rke2.tfvars.example
Comment (mdrahman-suse) [inline config/rke2.tfvars.example:33]: Redact ami info

File: config/rke2.tfvars.example
Comment (mdrahman-suse) [inline config/rke2.tfvars.example:28]: redact ami info

File: config/.ssh/aws_key.pem.example
Comment (fmoral2) [inline config/.ssh/aws_key.pem.example:2]: tbh i dont a see a value , specifying a path to the user creating to add his key. 
we can just give the example without doing this.

Comment (fmoral2) [reply config/.ssh/aws_key.pem.example:2]: i prefer to use this key in a different place , even though its this path we use in the container, locally i like to use another path.

File: config/config.yaml.example
Comment (fmoral2) [inline config/config.yaml.example:1]: add the value as variable = {product}

File: config/config.yaml.example
Comment (fmoral2) [inline config/config.yaml.example:2]: same here

File: config/.ssh/aws_key.pem.example
Comment (fmoral2) [reply config/.ssh/aws_key.pem.example:2]: so if i like to do this, maybe others as well

File: config/config.yaml.example
Comment (aganesh-suse) [reply config/config.yaml.example:1]: isnt this supposed to be a yaml file format? we should be doing 'variable: value' in which case?

File: config/.ssh/aws_key.pem.example
Comment (aganesh-suse) [reply config/.ssh/aws_key.pem.example:2]: 2 reasons i added this file.
1) to have created the .ssh folder and checked the same in. 
2) To have the permission of the file preset to 600 - so we can simply rename the file and use the same. 
Without this permission, the test run fails - and asks us to have strict permissions. this is simply to avoid that headache.

File: config/config.yaml.example
Comment (aganesh-suse) [reply config/config.yaml.example:2]: same as previous comment. sticking with yaml format of `variable: value`

File: config/.ssh/aws_key.pem.example
Comment (fmoral2) [reply config/.ssh/aws_key.pem.example:2]: `If someone chooses to not use the same, they can of course. It is always a good practise to provide a solution-when it is possible, than let people figure out their own solutions.`

Exactly!! thats why i mentioned to add this as an "example" that were it could be in comments....not actually creating the folder....got it ?

File: config/config.yaml.example
Comment (fmoral2) [reply config/config.yaml.example:1]: why do you think use a var in the value is not a yaml ?

File: config/config.yaml.example
Comment (fmoral2) [reply config/config.yaml.example:2]: same as above

File: config/config.yaml.example
Comment (aganesh-suse) [reply config/config.yaml.example:1]: i think we both misunderstood each other. my problem was with using "=" instead of the ":" in a yaml. - since that is not yaml formatting. got where you are coming from now 👍

Comment (aganesh-suse) [reply config/config.yaml.example]: Fixed.

File: config/config.yaml.example
Comment (aganesh-suse) [reply config/config.yaml.example]: Fixed.

File: config/.ssh/aws_key.pem.example
Comment (aganesh-suse) [reply config/.ssh/aws_key.pem.example:2]: This has been added as an "example" file. 
I do not see harm in leaving this in there as an option, since there are not actual secrets shared. 
The main points being that the file permissions have been preset and one does not have to mount files in the lima environment. I do not feel there is a good enough reason to remove this file. Please feel free to not use the same.

Comment (aganesh-suse) [reply config/.ssh/aws_key.pem.example]: Since we moved all the files into an examples folder - removing this .pem file example from this location.

**Review by rancher-max — APPROVED** [review]

File: modules/k3s/variables.tf
Comment (ShylajaDevadiga) [inline modules/k3s/variables.tf:18]: Is default required? Can we leave it blank?

**Review by ShylajaDevadiga — APPROVED** [review]
One small comment. Otherwise LGTM

File: modules/k3s/variables.tf
Comment (mdrahman-suse) [reply modules/k3s/variables.tf:18]: +1

**Review by mdrahman-suse — APPROVED** [review]
LGTM, once Shylaja's comment is addressed

**Comment by mdrahman-suse** [comment]
Also I see 8 commits, don't forget to squash your commits before merging!

File: modules/k3s/variables.tf
Comment (aganesh-suse) [reply modules/k3s/variables.tf]: fixed. removed this change.

**Review by rancher-max — APPROVED** [review]
LGTM -- when merging, select the "Squash and Merge" option

**Review by rancher-max — APPROVED** [review]

File: config/examples/k3s.tfvars
Comment (ShylajaDevadiga) [inline config/examples/k3s.tfvars:12]: What is password for?

**Review by ShylajaDevadiga — APPROVED** [review]

**Review by ShylajaDevadiga — APPROVED** [review]

**Review by rancher-max — APPROVED** [review]

**Review by mdrahman-suse — APPROVED** [review]
LGTM, dont forget to squash and merge

**Pull request merged**
